# Welcome to Intoduction to GWAS

This tutorial is part of the Bioinformatics for Biologists course at FU-Berlin. The assumption is that you have access to `evop-login.imp.fu-berlin.de` where the data is stored. The biggest part of GWAS tutorial was taken from [Marees et al. 2018](https://onlinelibrary.wiley.com/doi/full/10.1002/mpr.1608), and the majority of PGS tutorial was taken from [Choi et al. 2020](https://choishingwan.github.io/PRS-Tutorial/plink/).

This tutorial is broken down into several sections: 
<br/>- Introduction to GWAS 
<br/>- Introduction to plink 
<br/>- Introduction to datasets used in tutorial
<br/>- Quality control
<br/>- Population stratification
<br/>- Association analysis
<br/>- PGS

In addition, this tutorial provides explanation on data preparationo and prepocesing:
<br/>- Preparation of HapMap data for tutorial
<br/>- Preparation of 1KGP data for tutorial


At the end of this two-day tutorial, you should get familiar with plink and the basic workflow in genome wide association studies (GWAS), and with computing polygenic scores (PGS). 

During the first day, we will cover data QC for GWAS, and during the second day, we will perform the association analysis and compute PGS.